client/me/purchases/billing-history/billing-history-list-data-view.tsx

```diff
@@ -0,0 +1,66 @@
+import { Gridicon } from '@automattic/components';
+import { DataViews } from '@wordpress/dataviews';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import getPastBillingTransactions from 'calypso/state/selectors/get-past-billing-transactions';
+import isRequestingBillingTransactions from 'calypso/state/selectors/is-requesting-billing-transactions';
+import { useFieldDefinitions } from './hooks/use-field-definitions';
+import { usePagination } from './hooks/use-pagination';
+import { useReceiptActions } from './hooks/use-receipt-actions';
+import { useTransactionsFiltering } from './hooks/use-transactions-filtering';
+import { useTransactionsSorting } from './hooks/use-transactions-sorting';
+import { useViewStateUpdate } from './hooks/use-view-state-update';
+
+import 'calypso/components/dataviews/style.scss';
+import './style-data-view.scss';
+
+export interface BillingHistoryListProps {
+	getReceiptUrlFor: ( receiptId: string ) => string;
+}
+
+export default function BillingHistoryListDataView( {
+	getReceiptUrlFor,
+}: BillingHistoryListProps ) {
+	const transactions = useSelector( getPastBillingTransactions );
+	const isLoading = useSelector( isRequestingBillingTransactions );
+	const viewState = useViewStateUpdate();
+	const receiptActions = useReceiptActions( getReceiptUrlFor );
+
+	const actions = receiptActions.map( ( action ) => ( {
+		...action,
+		icon: <Gridicon icon={ action.iconName } />,
+	} ) );
+
+	const filteredTransactions = useTransactionsFiltering( transactions, viewState.view );
+	const sortedTransactions = useTransactionsSorting( filteredTransactions, viewState.view );
+	const { paginatedItems, totalPages, totalItems } = usePagination(
+		sortedTransactions,
+		viewState.view.page,
+		viewState.view.perPage
+	);
+	const translate = useTranslate();
+	const fields = useFieldDefinitions( transactions );
+	const defaultLayout = { table: {} };
```

Comment:
Thanks for moving this to its own variable, but by putting it inside a function, it's still recreated every time the function is called. Try moving this outside any function. That's a good idea any time the thing you're creating doesn't need to be recreated.